### PR TITLE
Lrs seggregation

### PR DESCRIPTION
**Pull request by rpkamp**
Fixes https://github.com/SURFnet/learning-analytics/issues/29

### CODE REVIEW

File: app/config/services.yml [hunk 1]
```diff
@@ -29,9 +24,20 @@ services:
         public: false
         arguments: ["@app.repo.recipe", "%iri%"]
 
+    app.lrs_factory:
+        class: AppBundle\LRS\LrsClientFactory
+        arguments:
+        - '@app.xapi_client_builder'
+        - '%lrs_credentials%'
```
Comment (seggen-ibuildings) [inline app/config/services.yml]: Indentation?

File: app/config/parameters.yml.dist [hunk 2]
```diff
@@ -37,3 +37,4 @@ parameters:
     simplesamlphp.attributes:
         email: urn.email
         fullName: urn.fullname
+        homeDomain: "urn:oid:1.3.6.1.4.1.25178.1.2.9"
```
Comment (seggen-ibuildings) [inline app/config/parameters.yml.dist]: Never mind. As discussed IRL: we are going to take the homeDomain as a parameter for promoting users to admin.

File: composer.json [hunk 3]
```diff
@@ -35,13 +35,13 @@
         "nelmio/cors-bundle": "^1.4",
         "php-xapi/client": "^0.4.0",
         "doctrine/doctrine-migrations-bundle": "^1.2",
+        "sensiolabs/security-checker": "^3.0.2",
```
Comment (seggen-ibuildings) [inline composer.json]: Is this intentional?

File: src/AppBundle/Repository/TrackCommandRepository.php [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace AppBundle\Repository;
+
+use AppBundle\Entity\TrackCommand;
+use Doctrine\ORM\EntityRepository;
+
+class TrackCommandRepository extends EntityRepository
+{
+    public function save(TrackCommand $command)
```
Comment (seggen-ibuildings) [inline src/AppBundle/Repository/TrackCommandRepository.php]: Why not `add()`?

File: tests/AppBundle/Controller/TrackingControllerTest.php [hunk 5]
```diff
@@ -32,6 +39,27 @@ public function setUp()
         $this->assertEquals(0, $exitCode, $output->fetch());
     }
 
+    private function mockCourseRepositoryIn(\Symfony\Bundle\FrameworkBundle\Client $client)
```
Comment (seggen-ibuildings) [inline tests/AppBundle/Controller/TrackingControllerTest.php]: Maybe import `Client`?

File: tests/AppBundle/Service/TrackingServiceTest.php [hunk 6]
```diff
@@ -122,17 +149,28 @@ public function testDelayedTracking()
         $cmd->setObject('lrs');
         $cmd->setVerb('tested');
 
-        $service = new TrackingService($mock, new StatementAssembler($repo, $this->getIris()), $em);
+        $courseRepositoryMock = $this->createMock(CourseRepository::class);
+        $courseRepositoryMock
+            ->expects($this->exactly(2))
+            ->method('get')
+            ->willReturn($this->getCourseMock());
```
Comment (seggen-ibuildings) [inline tests/AppBundle/Service/TrackingServiceTest.php]: The number of mocks and the complexity of setting up these mocks is a clear indication that the `TrackingService` needs refactoring. Do we have an item on the road map that would warrant this?

File: app/config/services.yml [hunk 7]
```diff
@@ -8,17 +8,12 @@ services:
         public: false
         calls:
             - [setBaseUrl, ["%lrs_url%"]]
-            - [setAuth, ["%lrs_username%", "%lrs_password%"]]
 
     app.xapi_client:
```
Comment (rpkamp) [inline app/config/services.yml]: Nope. Removed.

File: app/config/services.yml (hunk 1 above)
Comment (rpkamp) [inline app/config/services.yml]: Oops. Fixed.

File: composer.json (hunk 3 above)
Comment (rpkamp) [inline composer.json]: No. Fixed.

File: src/AppBundle/Repository/TrackCommandRepository.php (hunk 4 above)
Comment (rpkamp) [inline src/AppBundle/Repository/TrackCommandRepository.php]: That fits better with the Doctrine setup yes. Renamed.

File: tests/AppBundle/Controller/TrackingControllerTest.php (hunk 5 above)
Comment (rpkamp) [inline tests/AppBundle/Controller/TrackingControllerTest.php]: There already is a Client import, the LRS Client class.
I don't see a need to change this throughout the class.

File: tests/AppBundle/Service/TrackingServiceTest.php (hunk 6 above)
Comment (rpkamp) [inline tests/AppBundle/Service/TrackingServiceTest.php]: Not in the immediate future I think, no.

Comment (seggen-ibuildings) [inline tests/AppBundle/Service/TrackingServiceTest.php]: Pitty. Maybe if we change the MySQL table with a job queue... That would be a good time, I think.

**Pull request merged**